Python/sysmodule.c

```diff
@@ -2641,6 +2641,51 @@ sys__baserepl_impl(PyObject *module)
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+sys._clear_type_descriptors
+
+    type: object
+    /
+
+Private function for clearing certain descriptors from a type's dictionary.
+
+See gh-135228 for context.
+[clinic start generated code]*/
+
+static PyObject *
+sys__clear_type_descriptors(PyObject *module, PyObject *type)
+/*[clinic end generated code: output=7d5cefcf861909e0 input=5fdc23500d477de6]*/
+{
+    if (!PyType_Check(type)) {
+        PyErr_SetString(PyExc_TypeError, "argument must be a type");
+        return NULL;
+    }
+    PyTypeObject *typeobj = (PyTypeObject *)(type);
+    if (!_PyType_HasFeature(typeobj, Py_TPFLAGS_HEAPTYPE)) {
+        PyErr_SetString(PyExc_TypeError, "argument must be a heap type");
+        return NULL;
+    }
+    PyObject *dict = _PyType_GetDict(typeobj);
+    PyObject *dunder_dict = NULL;
+    if (PyDict_PopString(dict, "__dict__", &dunder_dict) < 0) {
```

Comment:
You can use `&_Py_ID(__dict__)` and `&_Py_ID(__weakref__)`.
```suggestion
    if (PyDict_Pop(dict, &_Py_ID(__dict__), &dunder_dict) < 0) {
```